packages/mantine/src/components/table/Table.types.ts

```diff
@@ -19,6 +19,10 @@ export interface TableLayoutProps<TData = unknown> {
      * @param datum the row for which the children should be generated.
      */
     getExpandChildren?: (datum: TData) => ReactNode;
+    /**
+     * Function that can be used to add additional attributes on rows
+     */
+    getRowAttributes?: (datum: TData) => Record<string, unknown>;
```

Comment:
I think the phrasing might be confusing, because the method says `getRowAttribute` where the definition says `add`. Could it be `setRowAttributes` ? Or were you thinking of it as getting the attribute passed from `datum: TData` and then we do logic with it ? (which I would be fine with)

Reply:
I named it `getRowAttributes` because from the table implementer standpoint you get the attributes from the row data. We already have the prop `getRowId` and `getExpandChildren` so it made sense to me to use the same naming pattern for the same kind of pattern:

```ts
getRowId: (row: TData) => string;
getRowAttributes: (row: TData) => Record<string, unknown>;
getExpandChildren: (row: TData) => ReactNode;
```

Reply:
Understandable, have a nice day

Reply:
![ink](https://github.com/coveo/plasma/assets/35579930/c262427f-ec0a-45e1-8eb0-c0cf250dd33f)